priceloop_api/priceloop/extras/table.py

```diff
@@ -0,0 +1,855 @@
+                col_lst.append(Column(c, ColumnType(COLUMNS_TYPES_MAPPING[df[c].dtype.name])))
+
+        obj = cls(workspace_name, table_name, col_lst)
+        obj.set_df(df)
+        return obj
+
+    def validate_df_types_against_table(self, df: pd.DataFrame) -> bool:
+        if len(df.columns) == len(self.columns):
+            for column in self.columns:
+                if column.name not in df.columns:
+                    raise ColumnsMismatch
+                else:
+                    if column.ctype == "json":
+                        try:
+                            df[column.name].apply(json.loads)
+
+                        except json.decoder.JSONDecodeError:
+                            raise ColumnTypesMismatch
+                    else:
+                        try:
+                            assert column.ctype == ColumnType(
+                                COLUMNS_TYPES_MAPPING[df[column.name].dtype.name]
+                            ), ColumnTypesMismatch
+                        except KeyError:
+                            raise
+            return True
+        else:
+            raise ColumnsMismatch
+
+    def set_df(self, df: pd.DataFrame, validate_values: bool = True) -> None:
+        self.validate_df_types_against_table(df)
+        if validate_values:
+            for column in df.columns:
+                if infer_dtype(df[column]).startswith("mixed"):
+                    raise ColumnMixedTypes(f"column {column} has mixed types")
+        self.df = df
+
+    def __init__(
+        self,
+        workspace_name: str,
+        table_name: str,
+        columns: List[Column],
+    ) -> None:
+        self.workspace = workspace_name
+        self.name = table_name
+        self.columns = columns
+
+    def initialize_table(
+        self,
+        client: AuthenticatedClient,
+    ) -> bool:
+        return TableUtils.initialize_table(self.workspace, self.name, self.columns, client)
+
+    def is_table_on_platform(self, client: AuthenticatedClient) -> bool:
+        return TableUtils.is_table_on_platform(self.workspace, self.name, client)
+
+    def read_table(
+        self,
+        client: AuthenticatedClient,
+        columns: Optional[List[str]] = None,
+        exclude_formulas: bool = False,
+    ) -> pd.DataFrame:
+        df = TableUtils.read_table(self.workspace, self.name, client, columns, exclude_formulas)
+
+        meta_data = TableUtils.get_metadata(self.workspace, self.name, client, exclude_formulas)
+        
+        columns_data = meta_data.columns
+        if isinstance(columns_data, Unset):
+            raise PlatformError
+
+        for column in columns_data:
+            if PLATFORM_TO_PANDAS_MAPPING.get(column.tpe):
+                df[column.name].astype(PLATFORM_TO_PANDAS_MAPPING[column.tpe])
+        return df
+
+    def __repr__(self) -> str:
+        return f"Table Object <name: {self.name} columns count: {self.columns_count}>"
+
+@attr.s(auto_attribs=True)
+class TableUtils:
```

Comment:
I'd say we put it in an extra module as well and use the newer `@define` attrs api